src/OpenCV/VideoIO/VideoWriter.hs

```diff
@@ -0,0 +1,148 @@
+{-# LANGUAGE QuasiQuotes #-}
+{-# LANGUAGE TemplateHaskell #-}
+{-# LANGUAGE RecordWildCards #-}
+
+module OpenCV.VideoIO.VideoWriter
+  ( VideoWriter
+  , VideoWriterSink(..)
+
+  , newVideoWriter
+  , videoWriterOpen
+  , videoWriterRelease
+  , fourCC
+  , videoWriterIsOpened
+  , videoWriterWrite
+--   , videoWriterGrab
+--   , videoWriterRetrieve
+  ) where
+
+import "base" Data.Int ( Int32 )
+import "base" Foreign.C.String ( withCString )
+import "base" Foreign.ForeignPtr ( ForeignPtr, withForeignPtr )
+import "base" Foreign.Marshal.Utils ( fromBool, toBool )
+import "linear" Linear.V2 ( V2(..) )
+import qualified "inline-c" Language.C.Inline as C
+import qualified "inline-c" Language.C.Inline.Unsafe as CU
+import qualified "inline-c-cpp" Language.C.Inline.Cpp as C
+import "this" OpenCV.Core.Types
+-- import "this" OpenCV.Core.Types.Mat
+import "this" OpenCV.Internal
+import "this" OpenCV.Internal.Exception
+import "this" OpenCV.Internal.C.Inline ( openCvCtx )
+import "this" OpenCV.Internal.C.Types
+-- import "this" OpenCV.Internal.Core.Types.Mat
+-- import "this" OpenCV.Internal.Core.Types.Size
+import "this" OpenCV.TypeLevel
+import "transformers" Control.Monad.Trans.Except ( ExceptT(ExceptT) )
+
+--------------------------------------------------------------------------------
+
+C.context openCvCtx
+
+C.include "opencv2/core.hpp"
+C.include "opencv2/videoio.hpp"
+C.using "namespace cv"
+
+--------------------------------------------------------------------------------
+
+newtype VideoWriter = VideoWriter {unVideoWriter :: ForeignPtr (C VideoWriter)}
+
+type instance C VideoWriter = C'VideoWriter
+
+instance WithPtr VideoWriter where withPtr = withForeignPtr . unVideoWriter
+
+instance FromPtr VideoWriter where
+    fromPtr = objFromPtr VideoWriter $ \ptr ->
+                [CU.exp| void { delete $(VideoWriter * ptr) }|]
+
+data VideoWriterSink
+   = VideoFileSink
+      { vfsFilePath  :: !FilePath
+      , vfsFourCC    :: !String
```

Comment:
Ah yea, that would be cool :-) But is every string a valid FourCC? What about the empty string? Or a string with 5 characters?

Maybe the function that converts a string to FourCC should be of type `String -> Maybe FourCC`.

For now I would simply put it in the VideoWriter module. It can always be moved if it is also needed for other things.